### ISSUE TITLE
Accelerator mismatch in AWS 

### ISSUE THREAD

**Issue by prashanth5192** [Accelize/HyperFiRe#1]
When I run /opt/accelize/accelerator/accelerator -m 0 -i samples/corpus.csv on aws with the corresponding Hyperfire AMI image I get,

Input DATA file 'samples/corpus.csv' is now open and its size is 50.2 KB (51445 B).
Output DATA file 'output.dat' is now open and its size is currently 0.
[Error]: Json Configuration File UDID "0a1fb394-3d3d-4a8e-b6ce-1aa3894b1b67" and Accelerator UDID "ffffffff-ffff-ffff-ffff-ffffffffffff" are not compatible for accelerator "RegexAxonerve_v20_DRM23_hfix2".
Failed to open accelerator
Output DATA file 'output.dat' is now close and its size is 0 B (0 B).
Input DATA file 'samples/corpus.csv' is now close.

Same thing happens when I use the run_example.py,

Exception apyfal.exceptions.ClientRuntimeException: ClientRuntimeException('Accelerator Client Error: \'sudo /opt/accelize/accelerator/accelerator -m 2\\n\\x1b[0;37mOutput DATA file \\\'output.dat\\\' is now open and its size is currently 0.\\x1b[0;m\\n\\x1b[0;31m[Error]: Json Configuration File UDID "0a1fb394-3d3d-4a8e-b6ce-1aa3894b1b67" and Accelerator UDID "ffffffff-ffff-ffff-ffff-ffffffffffff" are not compatible for accelerator "RegexAxonerve_v20_DRM23_hfix2".\\x1b[0m\\n\\x1b[1;91mFailed to open accelerator\\x1b[0;m\\n\\x1b[0;37mOutput DATA file \\\'output.dat\\\' is now close and its size is 0 B (0 B).\\x1b[0;m\\n\\x1b[0;41m==> FAILURE <==\\x1b[0;m\\n\\x1b[0;37mCreated output INFO file \\\'output.json\\\': Success\\x1b[0;m\\n\'',) in <bound method SysCallClient.__del__ of <apyfal.client.syscall.SysCallClient accelerator='axonerve_hyperfire' type='None'>> ignored

**Comment by acamposAccelize** [comment]
Hi @prashanth5192,
I'm suspecting you are using an old version of apyfal.
Can you please check that this command line returns version 1.1.0
` python -c 'import apyfal; print apyfal.__version__;'
`

If it is not the case please update using 
`pip install apyfal[all] -U
`
We will update the AMI to  support this old version in the next few weeks.

Thanks for the feedback.
Let us now if upgrading the apyfal module fix your issue.

**Comment by acamposAccelize** [comment]
All look right, from the information I see.
 - You are using the right AMI from the output.json version returned.
 - You were able to unlock the FPGA devise successfully at 7:41 p.m
Of course the result is not the one expected :-(

I need more information to investigate : 
- Commands or script started :
- OS:
- python version : 

My best guest is that you may in a stuck situation related to the previous issue. I'm highly suggest restarting a fresh instance using the "./run_example.py", to start from a clean environment.
Please provide the output of the script for debugging purpose.

I'm waiting for your feedback.
Best,
Alex

**Comment by prashanth5192** [comment]
Python : 2.7.5
Client OS: Centos 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
Host: AWS Hyperfire Image
Config:

;Apyfal configuration file
;=========================

;The configuration file is used to provide configuration information to Apyfal.

;Use of this file is optional. All parameters can also be
;passed as arguments to Apyfal classes and functions.

;All parameters or section in this file can be removed without issue.
;Non specified or missing parameters use default values.
;See your targeted accelerator’s documentation for information about
;which parameters are needed for the accelerator and host.

;The API automatically searches for "accelerator.conf"
;in the current working directory, or in the current user home directory.

;A custom path to a configuration file can also be passed to classes or with the
;``APYFAL_CONFIG_FILE`` environment variable.

;Read the "Getting Started" documentation for examples of parameters use.

;This configuration file can set any parameter as ``parameter =value`` in the
;related section, even if not specified in this example file.
;Note that parameter names are case sensitive.

;Subsections
;-----------

;Some sections support *subsections*.

;A subsection is a section that overrides parameters of a parent section.
;If a parameter is not specified in a subsection,
;it uses the parameter values of its parent.

;Subsection have the following format ``[section.subsection]``
;(With ``[section]`` meaning the parent section of this subsection).

;See each section’s documentation to see if subsections are supported
;and how to use them.

[accelize]
;---------------------------
;This section sets the account details for the
;Accelize server and the access keys created via:
;https://accelstore.accelize.com/user/applications

;Client and Secret ID from Accelize account page.

client_id = **Masked**

secret_id = **Masked**

[host]
;---------------------------
;This section contains all the information related to the host
;used to deploy the accelerator.

;This section support subsections based on ``host_type`` parameter.
;You can add parameters to the ``[host.host_type]`` subsection to
;override the ``[host]`` section parameters for the specified ``host_type``.

;Host type
;
;*Possible values:* ``Alibaba``, ``AWS``, ``OpenStack``, ``OVH``, ``localhost``
;
;``localhost`` is only allowed on Accelerator itself.
;
host_type = AWS

;Host AccessKey/account details for cloud hosts.
;
client_id = **Masked**

secret_id = **Masked**

;Host region for cloud hosts
;A region that supports instances with FPGA devices is required.
;
;*Required only for: Alibaba, AWS, OpenStack, OVH*
;
;*Possible values for:*
;
;- *Alibaba:* ``cn-hangzhou``
;- *AWS:* ``eu-west-1``, ``us-east-1``, ``us-west-2``
;- *OVH:* ``GRA3``
;
region = us-east-1

;OpenStack Project ID
;
;*Required only for: OpenStack, OVH*
;

;OpenStack Auth-URL
;
;*Required only for: OpenStack*
;
auth_url =

;IP address to use as to define accelerator URL.
;
;*Possible values:*
;
;- ``True`` *= Use private IP by default.*
;- ``False`` *= Use public IP by default.*
;
use_private_ip = False

;SSL certificate. Used to:
; - Allows to access accelerator using HTTPS instead of HTTP when configuring a
;   new host.
; - Allows client to verify HTTPS connection using this certificate.
;
ssl_cert_crt =

;Generates a self signed certificate and save it on path specified by
;``ssl_cert_crt`` and ``ssl_cert_key`` before configuring host.
;
;*Possible values:* ``True``, ``False``
;
ssl_cert_generate =

;Defining host side configuration
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

;SSL certificate private key. Allows to access accelerator using HTTPS instead
;of HTTP  when configuring a new host.
;
ssl_cert_key =

;Configuration file to use host side.
;
init_config =

;Bash script to execute on instance startup.
;
init_script =

;Using a Pre-Existing Host Instance
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

;Host ``client_id`` and ``secret_id`` are optional
;if one of these values is provided.

;Instance ID of the cloud host instance to use.
;
instance_id = i-0805d60a079c5cf81

;IP address of the host to use.
;
host_ip = 34.227.161.10

;Host instance stopping Behavior
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

;Defines the way the host instance is to be stopped
;in case of object closure, a script end or a function stop call.
;
;*Possible values:*
;
;- ``term`` *= Instance will be deleted.*
;- ``stop`` *= Instance will be stopped and can be restarted.*
;- ``keep`` *= Instance will stay in running mode.*
;
;*Optionally, override the following default values:*
;
;- *term: if a new instance has been started*
;- *keep: if using a pre-existing instance*
;
stop_mode = keep

;Overriding default host environment value
;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

;All of these parameters are optional.

;Prefix to add before instance name
;
host_name_prefix =

;Host Key pair to use
;
key_pair = aws.pem

;Host Security group to use
;
security_group = sg-00052e64f2f8e271c

;Authorization role (AWS IAM role/Alibaba RAM role).
;Generated to allow an instance to load FPGA bitstream and access to storage
;
;*Only for: Alibaba, AWS*
;
role =

;Authorization policy (AWS IAM policy/Alibaba RAM policy).
;Generated to allow an instance to load FPGA bitstream and access to storage
;
;*Only for: Alibaba, AWS*
;
policy =

[configuration]
;---------------------------

;This section enables *specific configuration parameters* for
;the accelerator configuration step to be passed.
;These parameters will act as default values.
;Any parameter passed to the “start” method will override the
;values defined here.

;Read your targeted accelerator’s documentation for information
;about the possible specific parameter values.

;You can add parameters to the ``[configuration.accelerator]`` subsection
;to override the ``[configuration]`` section parameters for the specified
;accelerator.

;This section also enables parameters that use the *JSON parameters file*
;format to be passed using the following ``parameters`` value:
;
parameters =

**Comment by acamposAccelize** [comment]
Hi @prashanth5192 ,
 
From the error message you got, it looks like you are running the accelerator on FPGA host without the "sudo" command.

Can you confirm my assumption?

**Comment by acamposAccelize** [comment]
Hi again, 
After our call, indeed the local issue was related to the missing "sudo" before the commands line.
In your case you were trying to start instance manually and then execute command inside the instance
In this case the hard way procedure is this one :
```
#Start instance using AWS console
#SSH to it
ssh -Yt -i ${key_pair} centos@${host_ip}
#Clone the repo to have sample files
git clone https://github.com/Accelize/HyperFiRe
cd HyperFiRe
#Add your Accelize credential
sudo vi /etc/accelize/credentials.json

#Add the right the fpga image to load 
sudo sh -c ' echo "fpgaimage = agfi-0af63b56a0c39dd47" > /etc/sysconfig/meteringclient'

#Load the accelerator
sudo systemctl start accelerator

#Unlock the accelerator
sudo systemctl start meteringclient
sudo systemctl start meteringsession

#Configure hyperfire
sudo /opt/accelize/accelerator/accelerator -m 0 -i samples/corpus.csv

#Process a file
sudo /opt/accelize/accelerator/accelerator -m 1 -i samples/shakespeare.txt -o results/shakespeare_out.txt

#Close the session
sudo systemctl stop meteringclient
sudo systemctl stop meteringsession
```
We will provide a simple way using apyfal in local mode (version 1.2.0 ) to execute the accelerator with the same commands used for the remote but in local mode. Purpose is to hide the complex steps below.



Regarding the issue you got from the remote execution , it was related to pycurl dependencies (not up-to-date).
We already detect that point corrected in version apyfal 1.2.0
